quickwit/quickwit-serve/src/jaeger_api/rest_handler.rs

```diff
@@ -0,0 +1,315 @@
+    responses(
+        (status = 200, description = "Successfully fetched traces information.", body = JaegerResponseBody )
+    ),
+    params(
+        TracesSearchQueryParams,
+        ("service" = Option<String>, Query, description = "The service name."),
+        ("start" = Option<i64>, Query, description = "The start time."),
+        ("end" = Option<i64>, Query, description = "The end time."),
+    )
+)]
+pub fn jaeger_traces_search_handler(
+    jaeger_service_opt: Option<JaegerService>,
+) -> impl Filter<Extract = (impl warp::Reply,), Error = Rejection> + Clone {
+    warp::path!("traces")
+        .and(warp::get())
+        .and(serde_qs::warp::query(serde_qs::Config::default()))
+        .and(require(jaeger_service_opt))
+        .then(jaeger_traces_search)
+        .map(|result| make_jaeger_api_response(result, BodyFormat::default()))
+}
+
+#[utoipa::path(
+    get,
+    tag = "Jaeger Traces",
+    path = "/otel-traces-v0_6/jaeger/api/traces/{id}/",
+    responses(
+        (status = 200, description = "Successfully fetched traces information for the provided id.", body = JaegerResponseBody )
+    )
+)]
+pub fn jaeger_traces_handler(
+    jaeger_service_opt: Option<JaegerService>,
+) -> impl Filter<Extract = (impl warp::Reply,), Error = Rejection> + Clone {
+    warp::path!("traces" / String)
+        .and(warp::get())
+        .and(require(jaeger_service_opt))
+        .then(jaeger_get_trace_by_id)
+        .map(|result| make_jaeger_api_response(result, BodyFormat::default()))
+}
+
+async fn jaeger_services(
+    jaeger_service: JaegerService,
+) -> Result<JaegerResponseBody<Vec<String>>, JaegerError> {
+    let get_services_response = jaeger_service
+        .get_services(with_tonic(GetServicesRequest {}))
+        .await
+        .unwrap()
+        .into_inner();
+    Ok(JaegerResponseBody::<Vec<String>> {
+        data: get_services_response.services,
+    })
+}
+
+async fn jaeger_service_operations(
+    service_name: String,
+    jaeger_service: JaegerService,
+) -> Result<JaegerSearchBody, JaegerError> {
+    let get_operations_request = GetOperationsRequest {
+        service: service_name,
+        span_kind: "".to_string(),
+    };
+    let get_operations_response = jaeger_service
+        .get_operations(with_tonic(get_operations_request))
+        .await
+        .unwrap()
+        .into_inner();
+
+    Ok(JaegerSearchBody {
+        data: Some(
+            get_operations_response
+                .operations
+                .into_iter()
+                .map(|op| op.name)
+                .collect_vec(),
+        ),
+    })
+}
+
+async fn jaeger_traces_search(
+    search_params: TracesSearchQueryParams,
+    jaeger_service: JaegerService,
+) -> Result<JaegerSearchBody, JaegerError> {
+    let query = TraceQueryParameters {
+        service_name: search_params.service.unwrap_or_default(),
+        operation_name: "stage_splits".to_string(),
```

Comment:
By looking at jaeger http requests, you can either specify operation "All" or the name of one operation. In the last case here is the HTTP request:

"http://localhost:16686/api/traces?end=1701555639361000&limit=20&lookback=3h&maxDuration&minDuration&operation=find_trace_ids&service=quickwit&start=1701544839361000"